velox/exec/MergeJoin.cpp

```diff
@@ -35,7 +35,8 @@ MergeJoin::MergeJoin(
       numKeys_{joinNode->leftKeys().size()},
       joinNode_(joinNode) {
   VELOX_USER_CHECK(
-      joinNode_->isInnerJoin() || joinNode_->isLeftJoin(),
+      joinNode_->isInnerJoin() || joinNode_->isLeftJoin() ||
```

Comment:
should we have a check somewhere to ensure that if this is a semi left join, we don't have any right side projections?

Reply:
@pedroerp Yes. We need this check. Added [here](https://github.com/facebookincubator/velox/pull/9877/files#diff-8279da91a596a4b97a73bc4d9aa635da1baaf254dcdb2f7f8049bd21e7a9551eR76).